statsmodels/genmod/generalized_linear_model.py

```diff
@@ -257,6 +285,17 @@ def _check_inputs(self, family, offset, exposure, endog):
             if offset.shape[0] != endog.shape[0]:
                 raise ValueError("offset is not the same length as endog")
 
+        if freq_weights is not None:
+            if freq_weights.shape[0] != endog.shape[0]:
+                raise ValueError("freq weights not the same length as endog")
+            if len(freq_weights.shape) > 1:
+                raise ValueError("freq weights has too many dimensions")
+
+        if self.freq_weights is None:
+            self.freq_weights = np.ones((endog.shape[0]))
+        if np.shape(self.freq_weights) == () and self.freq_weights > 1:
```

Comment:
I feel like that code was already in the project. At the time it was called data_weights, but the same check was used. Perhaps this was a check because generally, you assume that freq_weights are only discrete integers, and so this checked that freq_weights was a positive value? Just a guess.

Don't recall the actual reason beyond that.

> On Mar 8, 2016, at 6:31 PM, Josef Perktold <notifications@github.com> wrote:
> 
> In statsmodels/genmod/generalized_linear_model.py:
> 
> > @@ -257,6 +285,17 @@ def _check_inputs(self, family, offset, exposure, endog):
> >              if offset.shape[0] != endog.shape[0]:
> >                  raise ValueError("offset is not the same length as endog")
> >  
> > +        if freq_weights is not None:
> > +            if freq_weights.shape[0] != endog.shape[0]:
> > +                raise ValueError("freq weights not the same length as endog")
> > +            if len(freq_weights.shape) > 1:
> > +                raise ValueError("freq weights has too many dimensions")
> > +
> > +        if self.freq_weights is None:
> > +            self.freq_weights = np.ones((endog.shape[0]))
> > +        if np.shape(self.freq_weights) == () and self.freq_weights > 1:
> why self.freq_weights > 1? there is nothing specific about 1.
> ?
> 
> —
> Reply to this email directly or view it on GitHub.
>